### ISSUE TITLE
Undocumented backwards incompatibility in version 0.4.6

### ISSUE THREAD

**Issue by eXceediDeaL** [astropy/astroquery#2397]
Hello, I found that the most recent version **0.4.6** of astroquery probably introduces some **breaking changes**, which are not recorded in release notes.

1. Parameter renaming
     1. astroquery.utils.tap.taputils.get_suitable_output_file: isError -> is_error, outputFile -> output_file
     2. astroquery.utils.tap.xmlparser.utils.get_suitable_astropy_format: outputFormat -> output_format
     3. astroquery.utils.tap.xmlparser.utils.read_file_content: filePath -> file_path
     4. astroquery.utils.tap.xmlparser.utils.read_http_response: outputFormat -> output_format
2. `commons` and `modelutils` is not accessible from module `astroquery.esa.hubble.core`.
3. `DEVNULL` is not accessible from module `astroquery.utils.system_tools`.
4. Remove method `login_gui` from class `astroquery.esa.jwst.core.JwstClass`.
5. Add optional parameter `use_json` in method `astroquery.mast.services.ServiceAPI.service_request_async`.

To make it easier to understand the impact of the breaking changes, I write a short code snippet that reproduces the breaking changes.
The following code runs well in 0.4.5 but gets crashed in 0.4.6.

```python
# Change 1
astroquery.utils.tap.taputils.get_suitable_output_file(isError=True, outputFile='test.txt')
astroquery.utils.tap.xmlparser.utils.get_suitable_astropy_format(outputFormat='csv')
astroquery.utils.tap.xmlparser.utils.read_file_content(filePath='test.xml')
astroquery.utils.tap.xmlparser.utils.read_http_response(outputFormat='csv')
# causing TypeError: got an unexpected keyword argument 'outputFormat'

# Change 2-4
from astroquery.esa.hubble.core import commons, modelutils
from astroquery.utils.system_tools import DEVNULL

JwstClass().login_gui()
```

An example for Change 5, which would cause inconsistent signature if a subclass overrides the method.

```python
class SubServiceAPI(ServiceAPI):
  # The signature of this method has changed in 0.4.6 at base class ServiceAPI
  # but the subclass does not know the change.
  def service_request_async(self, service, params, page_size=None, page=None, **kwargs):
    return super().service_request_async(service, params, page_size, page, **kwargs)

service: ServiceAPI = SubServiceAPI(...)
service.service_request_async(..., use_json=True)
# use_json parameter will be accepted in kwargs of SubServiceAPI.service_request_async.
```

Maybe recording these changes in release notes helps to avoid user confusion during updating this version?

Kind regrads,

@eXceediDeaL

**Comment by bsipocz** [comment]
Hi @eXceediDeaL,

We usually take the changelog quite seriously, but mistakes can happen all the time. However, most of the examples you bring up was done correctly:

 > 
    Parameter renaming
        astroquery.utils.tap.taputils.get_suitable_output_file: isError -> is_error, outputFile -> output_file
        astroquery.utils.tap.xmlparser.utils.get_suitable_astropy_format: outputFormat -> output_format
        astroquery.utils.tap.xmlparser.utils.read_file_content: filePath -> file_path
        astroquery.utils.tap.xmlparser.utils.read_http_response: outputFormat -> output_format

utils.tap is a utility with the main purpose of serving the ESA modules. It has been noted at several places that it will be deprecated in favour of pyvo, and even then it's definitely not recommended or supported for end-users to use it.
In addition, as far as I see, the listed functions were never been part of any public API so changes to them should not be a concern, using non-public API happens at your own risk basis. 


>    commons and modelutils is not accessible from module astroquery.esa.hubble.core.

No, the only public API served by that module are, and always were stated to be `__all__ = ['ESAHubble', 'ESAHubbleClass']`. You should not import commons from their, just as you should not import numpy from an astropy namespace.

>    DEVNULL is not accessible from module astroquery.utils.system_tools.

`system_tools` is to provide utilities for astroquery internals, it's not part the API.


>    Remove method login_gui from class astroquery.esa.jwst.core.JwstClass.

Indeed. `jwst` was a new module added in 0.4.5, and it was an oversight that some of the cleanup that follow-ed never been documented in 0.4.6. However, I believe there wasn't really much data available at that point, so no actual users were around for `login_gui`. tagging @jespinosaar 

 >   Add optional parameter use_json in method astroquery.mast.services.ServiceAPI.service_request_async.

Indeed, that changes has been added as part of a huge PR, and we made an oversight of noting the new kwarg in the changelog. tagging @jaymedina 


So, in summary, I don't think any of the above should be the concern of the users. The two cases where we errenously omitted the changelog entries are for a branch new module that doesn't have users/data in the archive yet, while the second is the addition of extra functionality. Neither of those should cause concerns.

And the rest showcases that non-public API should not be relies on by users. It's a rule of thumb not just for astroquery but for any python packages.

Hope this summary helps.

**Comment by eXceediDeaL** [comment]
Yes, I notice that you write the changelog detailly, great job! I make some mistakes on the boundary of public API.
Thanks a lot for your complete reply and explanation on each change!
